### PR TITLE
[exn] Remove some uses of print

### PR DESCRIPTION
**Pull request by ejgallego**
Exceptions should not printed except for the top-level.

There is the weird anomaly-absorbing code in `Reductionops`, I wonder
how frequent that case is, but as the exception is absorbed printing
there could have a real impact.

### CODE REVIEW

File: pretyping/reductionops.ml
Comment (ejgallego) [inline pretyping/reductionops.ml:1094]: @ppedrot @maximedenes , I remember there was some discussion as to why we absorb anomalies here, would this a good moment to add a comment ?

Comment (ppedrot) [reply pretyping/reductionops.ml:1094]: The root issue is that tactics often send garbage to the kernel, e.g. asking whether two terms not having a common supertype are convertible. This breaks the conversion invariant, leading to random anomalies. I think it's indeed a good idea to write down a comment now.

Comment (ejgallego) [reply pretyping/reductionops.ml:1094]: Thanks for the comment @ppedrot , a last thing before I try a comment, should we consider tactics sending rubbish to the conversion algorithm a bug in the tactics?

Comment (ppedrot) [reply pretyping/reductionops.ml:1094]: I'd be inclined to believe so, but unfortunately it's not merely a uncommon behaviour. Tactics like `set` routinely break type-soundness internally, and I have no idea how to fix this without incurring a huge runtime penalty.

Comment (ejgallego) [reply pretyping/reductionops.ml:1094]: Yeah that's a complex matter, not sure tho the current status is good for people writing larguish bactracking stuff.

**Comment by ejgallego** [comment]
Umm, seems there is quite a bit of broken code that matches on `UserErr`

File: pretyping/reductionops.ml
Comment (gares) [reply pretyping/reductionops.ml:1102]: FTR, Looking for a head symbol as SSR does is already sufficient to rule out many nonsensical conversions. This filtering has very little runtime cost, but would clearly change the behaviour of vanilla set...

**Comment by ejgallego** [comment]
Bench just in case https://ci.inria.fr/coq/job/benchmark-part-of-the-branch/968

**Comment by gares** [comment]
The quick & async jobs fail by a genuine compilation error
```
File "topbin/coqproofworker_bin.ml", line 1:
Error: Required module `Reductionops' is unavailable
Makefile.build:429: recipe for target 'bin/coqproofworker.byte' failed
make[1]: *** [bin/coqproofworker.byte] Error 2
make[1]: *** Waiting for unfinished jobs....
```

**Comment by ejgallego** [comment]
> The quick & async jobs fail by a genuine compilation error

This is an old CI run, github was down most of the morning so it didn't update things properly.

**Comment by ejgallego** [comment]
> This is an old CI run, github was down most of the morning so it didn't update things properly.

That being said, this comment feature is worse than useless, as there is not way to determine if the current CI is still failing.

**Comment by Zimmi48** [comment]
> That being said, this comment feature is worse than useless, as there is not way to determine if the current CI is still failing on these jobs.

Note that it checks that the CI job is for the commit corresponding to the current PR's head. Not really my fault that GitHub was down most of this morning...

**Comment by Zimmi48** [comment]
If the current CI was still failing on those jobs, you would get a new comment from @coqbot.

**Comment by ejgallego** [comment]
> If the current CI was still failing on those jobs, you would get a new comment from @coqbot.

Still it is not possible to track the status unless one reads the very fine line regarding pushes to the branches; moreover, in case of a failing CI we get quite a few comment pollution. IMHO we should redesign this.

**Comment by gares** [comment]
I guess Coqbot should put its feedback in the checks window.
IIRC at the last call @Zimmi48 said this requires some extra work on Coqbot, which is scheduled but not done yet.

**Comment by ejgallego** [comment]
Bench https://ci.inria.fr/coq/job/benchmark-part-of-the-branch/970/console shows no change.

**Review by ppedrot — APPROVED** [review]

**Comment by ppedrot** [comment]
@ejgallego what about the CI failure? Is it relevant? Maybe you should perform a rebase to be sure before I merge.

**Comment by ejgallego** [comment]
> @ejgallego what about the CI failure? Is it relevant? Maybe you should perform a rebase to be sure before I merge.

The CI failure is spurios. I will rebase but no point now as still quite a few overlays for the persistent arrays PR have no been merged i think.

**Comment by ppedrot** [comment]
CI still quite unhappy, I will wait for the merge of missing overlays and a green CI before merging to be sure.

**Comment by ppedrot** [comment]
Since it's not possible lately to get a full CI, I think I'll merge this regardless of the CI status soon. Please voice up if you disagree.

**Comment by Zimmi48** [comment]
I think you can get an almost full CI at the moment if you want (just macOS and VST are failing). But I don't have any particular concern.

**Pull request merged**
